lnwallet/chancloser/chancloser.go

```diff
@@ -807,12 +815,23 @@ func (c *ChanCloser) ReceiveClosingSigned(msg lnwire.ClosingSigned) (
 		)
 		matchingSig := c.priorFeeOffers[remoteProposedFee]
 		if c.cfg.Channel.ChanType().IsTaproot() {
+			localWireSig, err := matchingSig.PartialSig.UnwrapOrErrV( //nolint:lll
+				fmt.Errorf("none local sig"),
+			)
+			if err != nil {
+				return noClosing, err
+			}
```

Comment:
same as previous point, 

I'd say this would be easier to read:

```
                        localWireSig, isNone := matchingSig.PartialSig.Unwrap()
			if err != nil {
				return noClosing, fmt.Errorf("none local sig"),
			}
```

Reply:
There certainly are a few ways to handle the `Some/None` cases, and this PR uses a newly added one that made these updates more straight forward IMO. I personally like the route where we directly return an error, as then it looks like the error handling we have everywhere else. 

Also I think your example should be: 
```go
                        localWireSig, isNone := matchingSig.PartialSig.Unwrap()
			if isNone {
				return noClosing, fmt.Errorf("none local sig"),
			}
```

Reply:
With the initial `WhenSome` we ensure that the wrapped variable isn't even accessible publicly if it's `None`. With things as is, the closure can add a bit of boiler plate at a call site though. This new API I added is arguably _less_ safe (can blow up if the caller doesn't check the second arg), but IMO it's consistent with error checking everywhere else, and we can use the liter to complain if the error value is ignored or not checked.

Reply:
This is the exact issue though. Is it `isNone` is it `isSome`? How do you know from just the type signature. Boolean return values are not a good idea most of the time. We can get away with it by making them the primary return value of a function named `isSomeThing` or `hasSomeOtherThing` because the context makes it very very clear. Here we just get `Unwrap` and it spits out a boolean. I don't want to have to read the godoc to know how to use the API. The API @Roasbeef suggests saves me having to read the docs to know how to use it.

Reply:
> Also I think your example should be:

yep - mistake. Updated the OG comment

Reply:
>  I personally like the route where we directly return an error, as then it looks like the error handling we have everywhere else.

But it isnt really like errors else where... we are passing in the error that we actually want to return to another func and then indirectly returning that. So to me it is less clear here what went wrong at error time.

> and we can use the liter to complain if the error value is ignored or not checked.

That is a good point yes

Reply:
> This is the exact issue though. Is it isNone is it isSome?

I think it is a well established pattern in go that if a func returns `A, bool` then that bool is the `ok` variable and so this would be `isSome` here

Reply:
> I don't want to have to read the godoc to know how to use the API

See reply above about `A, bool` being a well established pattern